src/vsphere_cpi/lib/cloud/vsphere/device_key_gen.rb

```diff
@@ -0,0 +1,15 @@
+module VSphereCloud
+  module DeviceKeyGenerator
+    def self.init(keys: nil)
```

Comment:
It looks like you want a singleton rather than a module here (if it's going to have an `init` method). Something like:

```ruby
require 'singleton'

module VSphereCloud
  class DeviceKeyGenerator
    def initialize(keys: nil)
      ...
    end
  end
end
```

Also are you sure you want the keyword argument `keys`? Seems like a regular argument is just as good here?